website/docs/guides/version-2-upgrade.html.md

```diff
@@ -0,0 +1,556 @@
+---
+layout: "aws"
+page_title: "Terraform AWS Provider Version 2 Upgrade Guide"
+sidebar_current: "docs-aws-guide-version-2-upgrade"
+description: |-
+  Terraform AWS Provider Version 2 Upgrade Guide
+---
+
+# Terraform AWS Provider Version 2 Upgrade Guide
+
+~> **NOTE:** This upgrade guide is a work in progress and will not be completed until the release of version 2.0.0 of the provider later this year.
+
+Upgrade topics:
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+- [data-source/aws_iam_role](#data-sourceaws_iam_role)
+- [data-source/aws_kms_secret](#data-sourceaws_kms_secret)
+- [data-source/aws_region](#data-sourceaws_region)
+- [resource/aws_api_gateway_api_key](#resourceaws_api_gateway_api_key)
+- [resource/aws_api_gateway_integration](#resourceaws_api_gateway_integration)
+- [resource/aws_api_gateway_integration_response](#resourceaws_api_gateway_integration_response)
+- [resource/aws_api_gateway_method](#resourceaws_api_gateway_method)
+- [resource/aws_api_gateway_method_response](#resourceaws_api_gateway_method_response)
+- [resource/aws_appautoscaling_policy](#resourceaws_appautoscaling_policy)
+- [resource/aws_autoscaling_policy](#resourceaws_autoscaling_policy)
+- [resource/aws_batch_compute_environment](#resourceaws_batch_compute_environment)
+- [resource/aws_cloudfront_distribution](#resourceaws_cloudfront_distribution)
+- [resource/aws_dx_lag](#resourceaws_dx_lag)
+- [resource/aws_ecs_service](#resourceaws_ecs_service)
+- [resource/aws_efs_file_system](#resourceaws_efs_file_system)
+- [resource/aws_instance](#resourceaws_instance)
+- [resource/aws_network_acl](#resourceaws_network_acl)
+- [resource/aws_redshift_cluster](#resourceaws_redshift_cluster)
+- [resource/aws_wafregional_byte_match_set](#resourceaws_wafregional_byte_match_set)
+
+<!-- /TOC -->
+
+## data-source/aws_iam_role
```

Comment:
This "data-source/name" style is not something I've seen us use anywhere else. I'd probably instead just write out `## aws_iam_role data source` to match how we usually talk about these things.